### PR TITLE
fixes for ThreadID on windows

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Fixes problems with Thread and some tests in concurrent on windows.
@marcusnaslund

### CODE REVIEW

File: source/concurrent/Thread.ooc [hunk 1]
```diff
@@ -64,9 +64,9 @@ version (unix || apple) {
 }
 
 version (windows) {
-	include windows
+	include inttypes
 
-	ThreadId: cover from DWORD {
+	ThreadId: cover from UInt {
```
Comment (sebastianbaginski) [inline source/concurrent/Thread.ooc]: According to windows docs, DWORD is 32 bit type. For some reasons it was typedef'ed to UInt64, so I needed to change it to UInt.

Comment (marcusnaslund) [inline source/concurrent/Thread.ooc]: You should probably change `DWORD: extern cover from Long` in `system/os/win32.ooc` instead.

Comment (sebastianbaginski) [inline source/concurrent/Thread.ooc]: Ah, so there it is:)

**Comment by sebastianbaginski** [comment]
Updated. Removed redundant DWORD typedef in Exception.

**Comment by marcusnaslund** [comment]
So you can define the same thing (differently) twice in global scope and rock goes "OK"...?

**Comment by marcusnaslund** [comment]
Ah, you need to either explicitly cast to `Int` (should work) or change to use `==` in the tests.

File: test/concurrent/WorkerThreadTest.ooc [hunk 2]
```diff
@@ -48,8 +48,8 @@ WorkerThreadTest: class extends Fixture {
 		this add("Correct threading", func {
 			mainThread := Thread currentThreadId()
 			for (i in 0 .. This threads count - 1) {
-				expect(mainThread != This threads[i])
-				expect(This threads[i], is equal to(This threads[i + 1]))
+				expect(mainThread equals(This threads[i]) == false)
+				expect(This threads[i] equals(This threads[i + 1]))
```
Comment (sebastianbaginski) [inline test/concurrent/WorkerThreadTest.ooc]: @marcusnaslund you meant this one ? I could not see the PRInt details correctly on windows

Comment (marcusnaslund) [inline test/concurrent/WorkerThreadTest.ooc]: I can't see the details either since it restarted, but yes, it seems to run fine now. :)

**Pull request merged**